stablehlo/reference/Ops.cpp

```diff
@@ -110,14 +103,16 @@ Tensor evalCosineOp(const Tensor &operand, Type resultType) {
 Tensor evalDynamicSliceOp(const Tensor &operand, ArrayRef<Tensor> startIndices,
                           ArrayRef<int64_t> sliceSizes, Type resultType) {
   Tensor result(resultType);
-  SmallVector<int64_t> adjustedStartIndices;
+  Index adjustedStartIndices(startIndices.size());
   for (size_t i = 0; i < startIndices.size(); ++i)
-    adjustedStartIndices.push_back(std::min(
-        std::max(startIndices[i].get({}).getIntegerValue().getSExtValue(), 0l),
-        operand.getType().getShape()[i] - sliceSizes[i]));
+    adjustedStartIndices[i] = std::min(
+        std::max(
+            startIndices[i].get(Index({})).getIntegerValue().getSExtValue(),
```

Comment:
It's unfortunate that we need additional boilerplate in the form of `Index({})`. Let's keep the `ArrayRef<int64_t>` overload as well?